tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/TimeRange.java

```diff
@@ -207,6 +220,38 @@ public boolean getRightClose() {
     return rightClose;
   }
 
+  /**
+   * Return the union of the given time ranges.
+   *
+   * @param unionCandidates time ranges already sorted in ascending order
+   * @return the union of time ranges
+   */
+  public static List<TimeRange> getUnions(ArrayList<TimeRange> unionCandidates) {
+    ArrayList<TimeRange> unionResult = new ArrayList<>();
+    Iterator<TimeRange> iterator = unionCandidates.iterator();
+    TimeRange range_curr;
+
+    if (!iterator.hasNext()) {
+      return unionResult;
+    } else {
+      TimeRange r = iterator.next();
+      range_curr = new TimeRange(r.getMin(), r.getMax());
+    }
+
+    while (iterator.hasNext()) {
+      TimeRange range_next = iterator.next();
+      if (range_curr.intersects(range_next)) {
+        range_curr.set(Math.min(range_curr.getMin(), range_next.getMin()),
+            Math.max(range_curr.getMax(), range_next.getMax()));
+      } else {
+        unionResult.add(new TimeRange(range_curr.getMin(), range_curr.getMax()));
```

Comment:
```suggestion
        unionResult.add(range_curr);
```

Reply:
No. Have you noticed that the next line changes the contents of the range_curr?

Reply:
Ok. I got it.